[Curl][Soup] Move empty functions of ResourceHandle[Curl|Soup] and SynchronousLoaderClient[Curl|Soup] to ResourceHandle and SynchronousLoaderClient

kshukuwa opened the pull request:
#### d8328394a067174adc2e2c93d7e41758eb78fa83
<pre>
[Curl][Soup] Move empty functions of ResourceHandle[Curl|Soup] and SynchronousLoaderClient[Curl|Soup] to ResourceHandle and SynchronousLoaderClient
<a href="https://bugs.webkit.org/show_bug.cgi?id=258117">https://bugs.webkit.org/show_bug.cgi?id=258117</a>

Reviewed by Michael Catanzaro.

The functions implemented in ResourceHandle[Curl|Soup].cpp and
SynchronousLoaderClient[Curl|Soup].cpp are not used in WebKit2,
so they are empty implementations and will not be added in the future.
Moving these functions reduces maintenance costs.  Also remove
unnecessary includes.

* Source/WebCore/loader/DocumentLoader.cpp:
* Source/WebCore/loader/FrameLoader.cpp:
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebCore/loader/PingLoader.cpp:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
* Source/WebCore/loader/cache/CachedResource.cpp:
* Source/WebCore/page/Chrome.cpp:
* Source/WebCore/platform/Curl.cmake:
* Source/WebCore/platform/SourcesSoup.txt:
* Source/WebCore/platform/network/ResourceHandle.cpp:
(WebCore::ResourceHandle::~ResourceHandle):
(WebCore::ResourceHandle::start):
(WebCore::ResourceHandle::cancel):
(WebCore::ResourceHandle::platformSetDefersLoading):
(WebCore::ResourceHandle::platformLoadResourceSynchronously):
(WebCore::ResourceHandle::shouldUseCredentialStorage):
(WebCore::ResourceHandle::didReceiveAuthenticationChallenge):
(WebCore::ResourceHandle::receivedCredential):
(WebCore::ResourceHandle::receivedRequestToContinueWithoutCredential):
(WebCore::ResourceHandle::receivedCancellation):
(WebCore::ResourceHandle::receivedRequestToPerformDefaultHandling):
(WebCore::ResourceHandle::receivedChallengeRejection):
(WebCore::ResourceHandle::platformContinueSynchronousDidReceiveResponse): Deleted.
* Source/WebCore/platform/network/ResourceHandle.h:
* Source/WebCore/platform/network/SynchronousLoaderClient.cpp:
(WebCore::SynchronousLoaderClient::didReceiveAuthenticationChallenge):
(WebCore::SynchronousLoaderClient::platformBadResponseError):
* Source/WebCore/platform/network/curl/ResourceHandleCurl.cpp: Removed.
* Source/WebCore/platform/network/curl/SynchronousLoaderClientCurl.cpp: Removed.
* Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp:
* Source/WebCore/platform/network/soup/ResourceHandleSoup.cpp: Removed.
* Source/WebCore/platform/network/soup/SynchronousLoaderClientSoup.cpp: Removed.
* Source/WebKit/NetworkProcess/Downloads/Download.h:
* Source/WebKit/NetworkProcess/Downloads/DownloadManager.h:
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
* Source/WebKit/NetworkProcess/soup/NetworkProcessSoup.cpp:
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:

Canonical link: <a href="https://commits.webkit.org/265228@main">https://commits.webkit.org/265228@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9bc83ef79ad832ab57b695128959b8f328423593

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10310 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10546 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10816 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11955 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9913 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10321 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12541 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10496 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12874 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10467 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11203 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8685 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12340 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8510 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9338 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16614 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9608 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9489 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12741 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9931 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8062 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9090 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2470 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13341 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9778 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
I would fill these all with `ASSERT_NOT_REACHED()` rather than remove those asserts, just to be on the safe side.

kshukuwa committed:
[Curl][Soup] Move empty functions of ResourceHandle[Curl|Soup] and SynchronousLoaderClient[Curl|Soup] to ResourceHandle and SynchronousLoaderClient
https://bugs.webkit.org/show_bug.cgi?id=258117

Reviewed by Michael Catanzaro.

The functions implemented in ResourceHandle[Curl|Soup].cpp and
SynchronousLoaderClient[Curl|Soup].cpp are not used in WebKit2,
so they are empty implementations and will not be added in the future.
Moving these functions reduces maintenance costs.  Also remove
unnecessary includes.

* Source/WebCore/loader/DocumentLoader.cpp:
* Source/WebCore/loader/FrameLoader.cpp:
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebCore/loader/PingLoader.cpp:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
* Source/WebCore/loader/cache/CachedResource.cpp:
* Source/WebCore/page/Chrome.cpp:
* Source/WebCore/platform/Curl.cmake:
* Source/WebCore/platform/SourcesSoup.txt:
* Source/WebCore/platform/network/ResourceHandle.cpp:
(WebCore::ResourceHandle::~ResourceHandle):
(WebCore::ResourceHandle::start):
(WebCore::ResourceHandle::cancel):
(WebCore::ResourceHandle::platformSetDefersLoading):
(WebCore::ResourceHandle::platformLoadResourceSynchronously):
(WebCore::ResourceHandle::shouldUseCredentialStorage):
(WebCore::ResourceHandle::didReceiveAuthenticationChallenge):
(WebCore::ResourceHandle::receivedCredential):
(WebCore::ResourceHandle::receivedRequestToContinueWithoutCredential):
(WebCore::ResourceHandle::receivedCancellation):
(WebCore::ResourceHandle::receivedRequestToPerformDefaultHandling):
(WebCore::ResourceHandle::receivedChallengeRejection):
(WebCore::ResourceHandle::platformContinueSynchronousDidReceiveResponse): Deleted.
* Source/WebCore/platform/network/ResourceHandle.h:
* Source/WebCore/platform/network/SynchronousLoaderClient.cpp:
(WebCore::SynchronousLoaderClient::didReceiveAuthenticationChallenge):
(WebCore::SynchronousLoaderClient::platformBadResponseError):
* Source/WebCore/platform/network/curl/ResourceHandleCurl.cpp: Removed.
* Source/WebCore/platform/network/curl/SynchronousLoaderClientCurl.cpp: Removed.
* Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp:
* Source/WebCore/platform/network/soup/ResourceHandleSoup.cpp: Removed.
* Source/WebCore/platform/network/soup/SynchronousLoaderClientSoup.cpp: Removed.
* Source/WebKit/NetworkProcess/Downloads/Download.h:
* Source/WebKit/NetworkProcess/Downloads/DownloadManager.h:
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
* Source/WebKit/NetworkProcess/soup/NetworkProcessSoup.cpp:
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:

Canonical link: https://commits.webkit.org/265228@main
